modules/tinymce/src/plugins/link/test/ts/browser/AttributesPostprocessTest.ts

```diff
@@ -0,0 +1,64 @@
+import { ApproxStructure, FocusTools, Waiter } from '@ephox/agar';
+import { describe, it, before, after } from '@ephox/bedrock-client';
+import { Fun } from '@ephox/katamari';
+import { SugarDocument } from '@ephox/sugar';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/link/Plugin';
+
+import { TestLinkUi } from '../module/TestLinkUi';
+
+describe('browser.tinymce.plugins.link.AttributesPostprocess', () => {
+  const newRel = Fun.constant('noopener noreferrer');
+  const attrsOverride = (attrs: any) => {
+    attrs.rel = newRel();
+  };
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'link',
+    toolbar: 'link',
+    link_attributes_postprocess: attrsOverride,
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ]);
+
+  before(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  after(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  it('TINY-11707: The link_attributes_postprocess can be used to override link attributes', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>Something</p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], ''.length, [ 0, 0 ], 'Something'.length);
+    await TestLinkUi.pOpenLinkDialog(editor);
+
+    FocusTools.setActiveValue(SugarDocument.getDocument(), 'http://www.tiny.cloud');
+    TestLinkUi.assertDialogContents({
+      href: 'http://www.tiny.cloud',
+      text: 'Something',
+      title: '',
+      target: ''
+    });
+    await TestLinkUi.pClickSave(editor);
+    await Waiter.pTryUntil('Waiting for content to have correct structure', () => {
```

Comment:
Is this really async?
Can we use assertContent instead?

There is a risk at least on the `href` attribute that we don't properly assert the right value since this assertContentStructure should really have been named `assertRawContentStructure` since that is what it does it takes the current DOM state and asserts that but for the `href` attribute for example there might be the cached `data-mce-href` value and that is what will be used when serializing. So does the postprocess properly update the `href` so that both the `href` and the `data-mce-href` attributes gets the right value?

We can still use the `assertContentStructure` for asserting both values but we then must remember to do so, so that is why the raw assertions like this is a bit more risky. It's easy to forget.